R/validate-pds-trips.R

```diff
@@ -0,0 +1,116 @@
+
+  logger::log_info("Validating pds trips...")
+  navigation_alerts <- validate_pds_navigation(pds_trips,hrs=hrs, km=km)
+
+  # take ready (?) columns
+  ready_cols <- pds_trips %>%
+    dplyr::select(.data$`Last Seen`,
+                  .data$IMEI,
+                  .data$Trip,
+                  .data$Ended)
+
+  validated_trips <-
+    list(navigation_alerts$validated_pds_duration,
+         navigation_alerts$validated_pds_distance) %>%
+    purrr::map(~ dplyr::select(.x,-alert_number)) %>%
+    purrr::reduce(dplyr::left_join) %>%
+    dplyr::left_join(ready_cols)
+
+  validated_trips_filename <- paste(pars$pds$trips$file_prefix,
+                                       "validated", sep = "_") %>%
+    add_version(extension = "rds")
+  readr::write_rds(x = validated_trips,
+                   file = validated_trips_filename,
+                   compress = "gz")
+  logger::log_info("Uploading {validated_trips_filename} to cloud sorage")
+  upload_cloud_file(file = validated_trips_filename,
+                    provider = pars$storage$google$key,
+                    options = pars$storage$google$options)
+  logger::log_success("File upload succeded")
+  }
+
+
+#' Validate pds trips duration and distance
+#'
+#' This function takes pds trips data and returns validated
+#' trip duration and distance.
+#'
+#' @param data A data frame containing pds trips
+#' @param hrs Limit of trip duration in hours to be considered a valid
+#' catch session.
+#' @param km Limit of trip distance traveled in Km to be considered a valid
+#' catch session.
+#'
+#' @return A list containing data frames with validated catch duration and
+#' catch distance traveled
+#'
+#' @export
+#' @importFrom rlang .data
+#'
+#' @examples
+#' \dontrun{
+#'   pars <- read_config()
+#'   pds_trips <- get_preprocessed_trips()
+#'   validate_pds_navigation(pds_trips)
+#' }
+#'
+validate_pds_navigation <- function(data,hrs =NULL, km=NULL){
+
+  validated_duration <- list(
+
+    validated_pds_duration = data %>%
+      dplyr::mutate(`Duration (Seconds)` = as.numeric(.data$`Duration (Seconds)`)) %>%
+      dplyr::transmute(alert_number=
+                         dplyr::case_when(.data$`Duration (Seconds)`> hrs*60^2 ~8 ,TRUE ~ NA_real_),#test if trip duration is longer than n hours
+                       `Duration (Seconds)`=
+                         dplyr::case_when(.data$`Duration (Seconds)` > hrs*60^2 ~ NA_real_,TRUE ~ .data$`Duration (Seconds)`),
+                       Trip=.data$Trip),
+
+    validated_pds_distance = data %>%
+      dplyr::mutate(`Distance (Meters)` = as.numeric(.data$`Distance (Meters)`)) %>%
+      dplyr::transmute(alert_number=
+                         dplyr::case_when(.data$`Distance (Meters)`> km*1000 ~ 9 ,TRUE ~ NA_real_),#test if trip distance is longer than n km
+                       `Distance (Meters)`=
+                         dplyr::case_when(.data$`Distance (Meters)` > km*1000 ~ NA_real_,TRUE ~ .data$`Distance (Meters)`),
```

Comment:
I think it would be good to also make NA the start and end time. The duration is just the difference between these two time points so if the duration is wrong, it is likely at least one of these two is wrong as well.